packages/frontend-2/components/project/page/models/UploadsDialog.vue

```diff
@@ -0,0 +1,207 @@
+        { id: 'file', header: 'File', classes: 'col-span-5' },
+        { id: 'size', header: 'Size', classes: 'col-span-2' },
+        { id: 'status', header: 'Status', classes: 'col-span-2' },
+        { id: 'date', header: 'Date', classes: 'col-span-2' },
+        {
+          id: 'actions',
+          header: '',
+          classes: 'col-span-1 flex items-center justify-end'
+        }
+      ]"
+      :items="items"
+      :loading="isVeryFirstLoading"
+      empty-message="This model has no uploads"
+      :max-height="300"
+    >
+      <template #file="{ item }">
+        <div
+          v-tippy="{
+            content: item.fileName.length > 35 ? item.fileName : undefined,
+            placement: 'top-start',
+            delay: 300
+          }"
+          class="truncate text-foreground"
+        >
+          {{ item.fileName }}
+        </div>
+      </template>
+      <template #size="{ item }">
+        <span class="text-foreground-2">{{ prettyFileSize(item.fileSize) }}</span>
+      </template>
+      <template #status="{ item }">
+        <CommonBadge
+          v-tippy="getStatusOptions(item).tooltip"
+          :color-classes="getStatusOptions(item).colorClasses"
+        >
+          {{ getStatusOptions(item).label }}
+        </CommonBadge>
+      </template>
+      <template #date="{ item }">
+        <span
+          v-tippy="formattedFullDate(item.convertedLastUpdate || item.uploadDate)"
+          class="text-foreground-2"
+        >
+          {{ formattedRelativeDate(item.convertedLastUpdate || item.uploadDate) }}
+        </span>
+      </template>
+      <template #actions="{ item }">
+        <FormButton
+          :icon-left="ArrowDownTrayIcon"
+          hide-text
+          size="sm"
+          color="outline"
+          @click="onDownload(item)"
+        />
+      </template>
+      <template #loader>
+        <InfiniteLoading
+          v-if="items?.length"
+          :settings="{ identifier }"
+          @infinite="onInfiniteLoad"
+        />
+      </template>
+    </LayoutTable>
+  </LayoutDialog>
+</template>
+<script setup lang="ts">
+import { ArrowDownTrayIcon } from '@heroicons/vue/24/outline'
+import {
+  FileUploadConvertedStatus,
+  fileUploadConvertedStatusLabels
+} from '@speckle/shared/blobs'
+import type { LayoutDialogButton } from '@speckle/ui-components'
+import { usePaginatedQuery } from '~/lib/common/composables/graphql'
+import { graphql } from '~/lib/common/generated/gql'
+import type { ProjectPageModelsUploadsDialog_FileUploadFragment } from '~/lib/common/generated/gql/graphql'
+import { useFileDownload } from '~/lib/core/composables/fileUpload'
+import { prettyFileSize } from '~~/lib/core/helpers/file'
+
+graphql(`
+  fragment ProjectPageModelsUploadsDialog_FileUpload on FileUpload {
+    id
+    convertedStatus
+    convertedMessage
+    fileName
+    fileSize
+    convertedLastUpdate
+    uploadDate
+    uploadComplete
+    branchName
+  }
+`)
+
+const getModelUploadsQuery = graphql(`
```

Comment:
better to load these only when modal opens, instead of using a fragment from the parent component's query, cause most of the time people aren't gonna look at these